lib/options.nix

```diff
@@ -391,6 +391,39 @@ rec {
       );
     };
 
+  mkOptionalPackage =
+    pkgs: name: package:
+    if lib.isOption package then
+      package
+    else if package != null then
+      lib.mkPackageOption pkgs name {
+        nullable = true;
+        default = package;
+      }
+    else
+      (lib.mkOption {
+        type = types.nullOr types.package;
+        description = ''
+          Package to use for ${name}.
+        '';
+      })
+      // {
+        _unpackaged = true;
+        _packageName = name;
+      };
```

Comment:
Does this survive the module system? I'm pretty sure particular parts of the module system that elaborate options will only inherit known attrs.

Reply:
From what I tested it does (as I managed to trigger the warnkngs